### ISSUE TITLE
Multi-Rate Channelizer Tests Broken

### ISSUE THREAD

**Issue by OrthogonalHawk** [OrthogonalHawk/libfalcon_dsp#25]
Output data comparison is incorrect

**Comment by OrthogonalHawk** [comment]
Found a fun thing with Python during the course of investigating this issue. Specifically, there appears to be a non-negligible difference between the **numpy.exp** and **numpy.cos** and **numpy.sin** function outputs as the input argument becomes large. The attached file shows a short Jupyter notebook that investigates the behavior of these functions and demonstrates that for large values their implementations differ more and more in calculated values. While this normally may not be an issue, for me in this unit test environment it matters quite a bit since I'm trying to demonstrate very similar results between Python and C++ and CUDA implementations. In the end, I've switched my Python implementation to explicitly use the **numpy.cos** and **numpy.sin** methods, which seem to agree more closely with the C++ and CUDA equivalents.

[PythonNumpyExponentialExperiment.pdf](https://github.com/OrthogonalHawk/libfalcon_dsp/files/4146229/PythonNumpyExponentialExperiment.pdf)

**Comment by OrthogonalHawk** [comment]
Finished making updates to fix this issue. Found a fun thing when I switched to using the **sin** and **cos** functions inside the frequency shift CUDA kernels from the **sinf** and **cosf** methods. Specifically, the double precision versions, now in use, seem to consume a substantial number of device resources. I had to either simplify the kernels and/or reduce the number of threads in a thread block before I could launch the frequency shift kernels without running into a "too many resources requested" CUDA error.
